test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1

```diff
@@ -1,64 +1,75 @@
 # Licensed under the MIT License.
-Describe 'ConvertTo-Json' -tags "CI" {
-    BeforeAll {
-        $newline = [System.Environment]::NewLine
-    }
 
+using namespace System.Environment
+using namespace System.Management.Automation
+using namespace Newtonsoft.Json.Linq
+
+Describe 'ConvertTo-Json' -tags 'CI' {
     It 'Newtonsoft.Json.Linq.Jproperty should be converted to Json properly' {
-        $EgJObject = New-Object -TypeName Newtonsoft.Json.Linq.JObject
-        $EgJObject.Add("TestValue1", "123456")
-        $EgJObject.Add("TestValue2", "78910")
-        $EgJObject.Add("TestValue3", "99999")
-        $dict = @{}
-        $dict.Add('JObject', $EgJObject)
-        $dict.Add('StrObject', 'This is a string Object')
-        $properties = @{'DictObject' = $dict; 'RandomString' = 'A quick brown fox jumped over the lazy dog'}
-        $object = New-Object -TypeName psobject -Property $properties
+        $object = [pscustomobject]@{
+            HashTable = [hashtable]@{
+                JObject = [JObject]::FromObject( @{
+                        TestValue1 = 123456;
+                        TestValue2 = 78910;
+                        TestValue3 = 99999
+                    } );
+                StrObject = 'This is a string Object'
+            };
+            RandomString = 'A quick brown fox jumped over the lazy dog'
+        }
+
         $jsonFormat = ConvertTo-Json -InputObject $object
         $jsonFormat | Should -Match '"TestValue1": 123456'
         $jsonFormat | Should -Match '"TestValue2": 78910'
         $jsonFormat | Should -Match '"TestValue3": 99999'
     }
 
-	It "StopProcessing should succeed" {
-        $ps = [PowerShell]::Create()
-        $null = $ps.AddScript({
-            $obj = [PSCustomObject]@{P1 = ''; P2 = ''; P3 = ''; P4 = ''; P5 = ''; P6 = ''}
-            $obj.P1 = $obj.P2 = $obj.P3 = $obj.P4 = $obj.P5 = $obj.P6 = $obj
-            1..100 | Foreach-Object { $obj } | ConvertTo-Json -Depth 10 -Verbose
-            # the conversion is expected to take some time, this throw is in case it doesn't
-            throw "Should not have thrown exception"
-        })
-        $null = $ps.BeginInvoke()
-        # wait for verbose message from ConvertTo-Json to ensure cmdlet is processing
-        Wait-UntilTrue { $ps.Streams.Verbose.Count -gt 0 }
-        $null = $ps.BeginStop($null, $null)
-        # wait a bit to ensure state has changed, not using synchronous Stop() to avoid blocking Pester
-        Start-Sleep -Milliseconds 100
-        $ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+    It 'StopProcessing should succeed' {
+        $ps = [powershell]::Create()
+        [void]$ps.AddScript( {
+                $obj = [pscustomobject]@{P1 = ''; P2 = ''; P3 = ''; P4 = ''; P5 = ''; P6 = ''}
+                $obj.P1 = $obj.P2 = $obj.P3 = $obj.P4 = $obj.P5 = $obj.P6 = $obj
+                (1..100).ForEach{
+                    Write-Verbose -Message 'Ready' -Verbose
```

Comment:
There's a race condition here where this message would cause line 43 to think that `ConvertTo-Json` has started processing, but the `BeginStop` online 46 could be called after this verbose message but before `ConvertTo-Json` actually started thus invalidating this test.  I believe `ConvertTo-Json -Verbose` will output some verbose message thus fulfilling the wait.

Reply:
ConvertTo-Json never writes to the verbose stream, and doesn't write to the success stream until it is done. For example:
`(1..5) | ConvertTo-Json -Verbose | ForEach-Object { $_; Pause }`
Note there is no verbose output written and that the Pause function is only called once.

In the original code Wait-UntilTrue times out because of this.

Reply:
@xtqqczze ok, I thought I had added a verbose message previously, but perhaps it got removed.  My preference would be to insert a verbose message at start of `convertto-json` as that guarantees the processing has started and makes this test valid.

Reply:
Issue opened: #9489